Unable to build: Quarkus ChainBuildException due to JFR dependency

### Describe the bug

In Quarkus version - 3.30.2

Ran command - ./gradlew clean build -x test

Logs-

> Task :quarkusAppPartsBuild FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':quarkusAppPartsBuild'.
> There was a failure while executing work items
   > A failure occurred while executing io.quarkus.gradle.tasks.worker.BuildWorker
      > io.quarkus.builder.ChainBuildException: Cycle detected:
                           io.quarkus.jfr.deployment.JfrProcessor#registerVersion produced class io.quarkus.arc.deployment.AdditionalBeanBuildItem
                        to io.quarkus.arc.deployment.ArcProcessor#initialize produced class io.quarkus.arc.deployment.ContextRegistrationPhaseBuildItem
                        to io.quarkus.arc.deployment.ArcProcessor#registerBeans produced class io.quarkus.arc.deployment.BeanDiscoveryFinishedBuildItem
                        to io.quarkus.grpc.deployment.GrpcClientProcessor#discoverInjectedClients produced class io.quarkus.deployment.builditem.FeatureBuildItem
                        to io.quarkus.jfr.deployment.JfrProcessor#registerVersion


### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

1. Create a new Quarkus project (Gradle or Maven), or use an existing one.

2. Add the following extensions to the project (if not already present):

```
quarkus-grpc-client

quarkus-jfr
```

Build the application:

`./gradlew build`


The build fails with the following exception:

```
Execution failed for task ':quarkusAppPartsBuild'.
io.quarkus.builder.ChainBuildException: Cycle detected:
JfrProcessor#registerVersion → ArcProcessor → GrpcClientProcessor → JfrProcessor#registerVersion
```

### Output of `uname -a` or `ver`

Darwin Priyas-MacBook-Pro.local 25.1.0 Darwin Kernel Version 25.1.0: Mon Oct 20 19:33:36 PDT 2025; root:xnu-12377.41.6~2/RELEASE_ARM64_T6030 arm64

### Output of `java -version`

OpenJDK Runtime Environment Corretto-21.0.8.9.1 (build 21.0.8+9-LTS) OpenJDK 64-Bit Server VM Corretto-21.0.8.9.1 (build 21.0.8+9-LTS, mixed mode, sharing)

### Quarkus version or git rev

Quarkus version - 3.30.2

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 8.8

### Additional information

_No response_

--- a/extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/GrpcClientProcessor.java
+++ b/extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/GrpcClientProcessor.java
@@ -115,9 +115,13 @@ public class GrpcClientProcessor {
     }
 
     @BuildStep
+    FeatureBuildItem grpcClientFeature() {
+        return new FeatureBuildItem(GRPC_CLIENT);
+    }
+
+    @BuildStep
     void discoverInjectedClients(BeanDiscoveryFinishedBuildItem beanDiscovery,
             BuildProducer<GrpcClientBuildItem> clients,
-            BuildProducer<FeatureBuildItem> features,
             CombinedIndexBuildItem index) {
 
         Map<String, GrpcClientBuildItem> items = new HashMap<>();
@@ -217,7 +221,7 @@ public class GrpcClientProcessor {
                 clients.produce(item);
                 LOGGER.debugf("Detected client associated with the '%s' configuration prefix", item.getClientName());
             }
-            features.produce(new FeatureBuildItem(GRPC_CLIENT));
+
         }
     }
 

--- a/integration-tests/opentelemetry-grpc/pom.xml
+++ b/integration-tests/opentelemetry-grpc/pom.xml
@@ -26,6 +26,11 @@
 
         <dependency>
             <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-jfr</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>io.quarkus</groupId>
             <artifactId>quarkus-rest</artifactId>
         </dependency>
         <dependency>
@@ -100,6 +105,19 @@
         <dependency>
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-opentelemetry-deployment</artifactId>
+            <version>${project.version}</version>
+            <type>pom</type>
+            <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-jfr-deployment</artifactId>
             <version>${project.version}</version>
             <type>pom</type>
             <scope>test</scope>

